webbeans-impl/src/main/java/org/apache/webbeans/config/BeansDeployer.java

```diff
@@ -1401,12 +1401,16 @@ else if (packageVetoed)
             {
                 String previousPackage = name.substring(0, idx);
                 Boolean result = packageVetoCache.get(previousPackage);
-                if (result != null && result)
+                if (result != null)
                 {
                     return result;
                 }
                 while (true)
                 {
+                    if (!previousPackage.isEmpty() && packageVetoCache.containsKey(previousPackage))
```

Comment:
basically fine, but this will check the uppermost package twice. It already got checked before entering the loop for the first time.
Being a performance critical spot I'd also prefer to do `Boolean result = packageVetoCache.get(previousPackage);` as containsKey and get are basically the same operation.

Reply:
Updated